x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/dashboards/data/dashboard_migrations_dashboards_client.ts

```diff
@@ -0,0 +1,128 @@
+} from '@elastic/elasticsearch/lib/api/types';
+import { MigrationTaskStatusEnum } from '../../../../../common/siem_migrations/model/common.gen';
+import type { SplunkOriginalDashboardExport } from '../../../../../common/siem_migrations/model/vendor/dashboards/splunk.gen';
+import { SiemMigrationStatus } from '../../../../../common/siem_migrations/constants';
+import { SiemMigrationsDataBaseClient } from '../../common/data/siem_migrations_data_base_client';
+import {
+  type DashboardMigrationDashboard,
+  type DashboardMigrationTaskStats,
+} from '../../../../../common/siem_migrations/model/dashboard_migration.gen';
+
+/* BULK_MAX_SIZE defines the number to break down the bulk operations by.
+ * The 500 number was chosen as a reasonable number to avoid large payloads. It can be adjusted if needed. */
+const BULK_MAX_SIZE = 500 as const;
+
+export class DashboardMigrationsDataDashboardsClient extends SiemMigrationsDataBaseClient {
+  /** Indexes an array of dashboards to be processed as a part of single migration */
+  async create(
+    migrationId: string,
+    originalDashboards: SplunkOriginalDashboardExport[]
+  ): Promise<void> {
+    const index = await this.getIndexName();
+    const profileId = await this.getProfileUid();
+
+    let originalDashboardsMaxBatch: SplunkOriginalDashboardExport[];
+    const createdAt = new Date().toISOString();
+    while ((originalDashboardsMaxBatch = originalDashboards.splice(0, BULK_MAX_SIZE)).length) {
+      await this.esClient
+        .bulk<DashboardMigrationDashboard>({
+          refresh: 'wait_for',
+          operations: originalDashboardsMaxBatch.flatMap(({ result: { ...originalDashboard } }) => [
+            { create: { _index: index } },
+            {
+              migration_id: migrationId,
+              '@timestamp': createdAt,
+              status: SiemMigrationStatus.PENDING,
+              created_by: profileId,
+              updated_by: profileId,
+              updated_at: createdAt,
+              original_dashboard: {
+                id: originalDashboard.id,
+                title: originalDashboard.label ?? originalDashboard.title,
+                description: originalDashboard.description ?? '',
+                data: originalDashboard?.['eai:data'],
+                format: 'xml',
+                vendor: 'splunk',
+                last_updated: originalDashboard.updated,
+                vendor_properties: {
+                  app: originalDashboard['eai:acl.app'],
+                  owner: originalDashboard['eai:acl.owner'],
+                  sharing: originalDashboard['eai:acl.sharing'],
+                },
+              },
+            },
+          ]),
+        })
+        .catch((error) => {
+          this.logger.error(
+            `Error adding dashboards to migration (id:${migrationId}) : ${error.message}`
+          );
+          throw error;
+        });
+    }
+  }
+
+  async getStats(migrationId: string): Promise<Omit<DashboardMigrationTaskStats, 'name'>> {
```

Comment:
nit: This function, and others in these data classes, look so similar to the ones in the rules data client classes, we could likely share the logic by putting these functions in SiemMigrationsDataBaseClient. The only difference between this stats function and the stats function in RuleMigrationsDataRulesClient is the dashboards key in the return + the status field.

Reply:
Additionally, i also think that `SiemMigrationsDataBaseClient` is very broad in scope such that it serves as base client for `migrations`, `rules/dashboards`, `resources` clients and this stats is just needed for `rules/dashboard` client.